### PR TITLE
osbuilder: Add USE_PODMAN as an alternate for USE_DOCKER

### PR DESCRIPTION
**Pull request by nitkon**
In case a user wants to use podman instead of
    docker to build initrd/rootfs images, facilitate
    it by setting the variable `USE_PODMAN=true`.
    
    Fixes: #370
    
    Signed-off-by: Nitesh Konkar <niteshkonkar@in.ibm.com>

### CODE REVIEW

**Comment by nitkon** [comment]
cc @jodh-intel @jcvenegas for review.

File: image-builder/image_builder.sh [hunk 1]
```diff
@@ -466,12 +467,20 @@ main() {
 		exit 0
 	fi
 
-	if [ -n "${USE_DOCKER}" ] ; then
-		build_with_docker "${rootfs}" "${image}" "${fs_type}" "${block_size}" \
-						  "${root_free_space}" "${agent_bin}" "${agent_init}"
+	if [ -n "${USE_DOCKER}" ]; then
```
Comment (devimc) [inline image-builder/image_builder.sh]: so, `USE_DOCKER` takes precedence over `USE_PODMAN`, it should be mentioned in the README

File: image-builder/image_builder.sh [hunk 2]
```diff
@@ -466,12 +467,20 @@ main() {
...
+		container_engine="docker"
+	elif [ -n "${USE_PODMAN}" ]; then
+		container_engine="podman"
+	fi
+
+	if [ -n "$container_engine" ]; then
+
```
Comment (devimc) [inline image-builder/image_builder.sh]: blank line ?

File: rootfs-builder/rootfs.sh [hunk 3]
```diff
@@ -124,6 +124,9 @@ ROOTFS_DIR          Path to the directory that is populated with the rootfs.
 
 USE_DOCKER          If set, build the rootfs inside a container (requires
                     Docker).
+
+USE_PODMAN          If set, build the rootfs inside a container (requires
```
Comment (devimc) [inline rootfs-builder/rootfs.sh]: again, `USE_DOCKER` takes precedence over `USE_PODMAN`

**Comment by nitkon** [comment]
Thanks @devimc , addressed the comments.
/retest

File: rootfs-builder/rootfs.sh [hunk 4]
```diff
@@ -124,6 +124,10 @@ ROOTFS_DIR          Path to the directory that is populated with the rootfs.
 
 USE_DOCKER          If set, build the rootfs inside a container (requires
                     Docker).
+		    Default value: <not set>
```
Comment (chavafg) [inline rootfs-builder/rootfs.sh]: indentation issue?

File: rootfs-builder/rootfs.sh [hunk 5]
```diff
@@ -124,6 +124,10 @@ ROOTFS_DIR          Path to the directory that is populated with the rootfs.
...
+
+USE_PODMAN          If set and USE_DOCKER not set, then build the rootfs inside
+		    a podman container (requires podman).
```
Comment (chavafg) [inline rootfs-builder/rootfs.sh]: indentation issue?

File: rootfs-builder/rootfs.sh [hunk 6]
```diff
@@ -308,17 +312,23 @@ build_rootfs_distro()
...
+			container_engine="docker"
+		elif [ -n "${USE_PODMAN}" ]; then
+			container_engine="podman"
+		fi
+
 		image_name="${distro}-rootfs-osbuilder"
 
 		generate_dockerfile "${distro_config_dir}"
-		docker build  \
+	"$container_engine" build  \
```
Comment (chavafg) [inline rootfs-builder/rootfs.sh]: indentation

File: rootfs-builder/rootfs.sh [hunk 7]
```diff
@@ -353,7 +363,7 @@ build_rootfs_distro()
 		#Make sure we use a compatible runtime to build rootfs
 		# In case Clear Containers Runtime is installed we dont want to hit issue:
 		#https://github.com/clearcontainers/runtime/issues/828
-		docker run  \
+	"$container_engine" run  \
```
Comment (chavafg) [inline rootfs-builder/rootfs.sh]: indentation

**Comment by jcvenegas** [comment]
@nitkon  could you fix the identation  ?

**Comment by nitkon** [comment]
Thanks @chavafg , all green

**Pull request merged**